File: backend/src/domain/clinic/ClinicBuilder.ts
import {Clinic} from "./Clinic";

export class ClinicBuilder {
  private id?: string
  private name?: string
  private address?: string
  private avatarUrl?: string

  setId(id: string) {
    this.id = id;
    return this;
  }

  setName(name: string) {
    this.name = name;
    return this;
  }

  setAddress(address: string) {
    this.address = address;
    return this;
  }

  setAvatarUrl(avatarUrl: string) {
    this.avatarUrl = avatarUrl;
    return this;
  }

  build() {
    return new Clinic(this.id, this.name, this.address, this.avatarUrl)
  }
}

File: backend/src/main/java/com/hypocrates/hypocrates/core/domain/clinic/ClinicBuilder.java
package com.hypocrates.hypocrates.core.domain.clinic;

import com.hypocrates.hypocrates.core.domain.patient.Patient;
import com.hypocrates.hypocrates.core.domain.staff.Staff;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.Set;

@Accessors(chain = true) @Setter
public class ClinicBuilder {
    private Long id;
    private String name;
    private String address;
    private String avatarUrl;
    private Set<Patient> patients = Set.of();
    private Set<Staff> staff = Set.of();
    private Clinic parentClinic;
    private Set<Clinic> childClinics = Set.of();

}
